src/Services/DataService/Epics/FetchRecordEpic.test.ts

```diff
@@ -0,0 +1,761 @@
+              type: "PUSH_RECORD",
+              payload: {
+                id,
+                fullText: expectedResult.fullText,
+              },
+            });
+            resolve();
+          } catch (e) {
+            reject(e);
+          }
+        }, DEFAULT_COALESCE_BUFFER_TIME + 1);
+      });
+    });
+
+  });
+
+  describe("performBufferedRequest", () => {
+
+    it("should emit a FETCH_ALL action if multiple items are requested within the given period", () => {
+      const firstId = random.number().toString();
+      const secondId = random.number().toString();
+      const thirdId = random.number().toString();
+      const fourthId = random.number().toString();
+      const firstTimeOutPeriod = DEFAULT_COALESCE_BUFFER_TIME + 1;
+      const secondTimeOutPeriod = DEFAULT_COALESCE_BUFFER_TIME + 2;
+
+      const performRequest = performBufferedRequest(context);
+      
+      const sut = performRequest(firstId);
+
+      let action;
+      sut
+        .take(1)
+        .subscribe(a => action = a);
+
+      performRequest(secondId);
+      performRequest(thirdId);
+
+      setTimeout(() => {
+        performRequest(fourthId);
+      }, firstTimeOutPeriod);
+
+      const queryBuilder = new QueryBuilder(context.name, { ids: [firstId, secondId, thirdId] });
+      return new Promise((resolve, reject) => {
+        setTimeout(() => {
+          try {
+            expect(action).to.deep.equal({
+              type: "FETCH_ALL",
+              payload: queryBuilder,
+            });
+            resolve();
+          } catch (e) {
+            reject(e);
+          }
+        }, secondTimeOutPeriod);
+      });
+    });
+
+    it("should return an empty observable when it's called subsequent times with the same context name within the timeout period", () => {
+      const id = random.number().toString();
+      const secondId = random.number().toString();
+
+      const performRequest = performBufferedRequest(context);
+      performRequest(id);
+
+      const spies = {
+        next: spy(),
+        complete: spy(),
+      };
+
+      performRequest(secondId)
+        .subscribe(spies);
+
+      expect(spies.next.callCount).to.equal(0, "it should not call next method when subscribing");
+      expect(spies.complete.callCount).to.equal(1, "it should complete immediately");
+    });
+
+    it("should return separate buffered observables for contexts of different names", () => {
+      const firstId = random.number().toString();
+      const secondId = random.number().toString();      
+      const timeoutPeriod = DEFAULT_COALESCE_BUFFER_TIME + 2;
+
+      const firstPerformRequest = performBufferedRequest(context);
+      const newContext = {
+        ...context,
+        name: random.word(),
+      };
+      const secondPerformRequest = performBufferedRequest(newContext);
+      
+      firstPerformRequest(firstId);
+      const sut = secondPerformRequest(secondId);
+      let action;
+      sut
+        .take(1)
```

Comment:
again, I dont think this take(1) is necessary for the performBufferedRequest() calls since there is already a take(1) in the method, but let me know if this is incorrect